CONTRIBUTING.md

```diff
@@ -97,127 +100,167 @@ $ sudo apt-get install cmake ninja-build
+It will show a message like "This branch is 1 commit ahead of `shader-slang/slang:master`."
+You can create a PR by clicking on the message.
 
-For the Pull Request, you will need to write a PR message. This message is for a set of commits you are requesting to pull. Try to make it brief because the actual details should be in the commit messages of each commit.
- 
-The PR requires an approval from people who have permissions. They will review the changes before approve the Pull. During this step, you will get feedbacks from other people and they may request you to make some changes. When you need to make adjustments, you can commit new changes to the branch in your forlked repository that already has the changes in PR process. When new commits are added to the branch, they will automatically appear in the PR.
+## Pull Request
+Once a PR is created against `shader-slang/slang:master`, the PR will be merged when the following conditions are met:
+1. The PR is reviewed and got approval.
+2. All of the workflows pass.
 
-## Addressing Code Reviews and Keep Branch In-Sync
-After your pull request is submitted, you will receive code reviews from the community within 24 hours. Follow-up changes that address review comments should be pushed to your pull request branch as additional commits. When your branch is out of sync with top-of-tree, submit a merge commit to keep them in-sync. Do not rebase and force push after the PR is created to keep the change history during the review process.
+When the conditions above are all met, you will have a chance to rewrite the commit message. Since the Slang repo uses the "squash" strategy for merging, multiple commits in your PR will become one commit. By default, GitHub will concatenate all of the commit messages sequentially, but often it is not readable. Please rewrite the final commit message in a way that people can easily understand what the purpose of the commit is.
 
-Use these commands to sync your branch:
+There are two cases where the workflow may fail for reasons that are not directly related to the change:
+1. "Breaking change" labeling is missing.
+2. Source code "Format" needs to be changed.
+
+### Addressing Code Reviews
+After your pull request is created, you will receive code reviews from the community within 24 hours.
+
+The PR requires approval from people who have permissions. They will review the changes before approving the pull. During this step, you will get feedback from other people, and they may request you to make some changes.
+
+Follow-up changes that address review comments should be pushed to your pull request branch as additional commits. Any additional commits made to the same branch in your forked repository will show up on the PR as incremental changes.
 
+When your branch is out of sync with top-of-tree, submit a merge commit to keep them in sync. Do not rebase and force push after the PR is created to keep the change history during the review process.
```

Comment:
Add:
"The Slang repository is configured to use "squashing" strategy when merging pull requests. Commits accumulated in your PR will be automatically squashed into a single commit to top of tree when it lands."

Reply:
This is already on the document at line 270.